src/app/shared/providers/services/auth.service.ts

```diff
@@ -63,8 +73,22 @@ export class AuthService {
       );
   }
 
+  refreshAccessToken(): Observable<AuthRefreshTokenResponse> {
+    const refreshToken: string | null = localStorage.getItem('refreshToken');
+
+    return this.#http.post<AuthRefreshTokenResponse>(`https://securetoken.googleapis.com/v1/token?key=${environment.apiKey}`, {
+      grant_type: 'refresh_token',
+      refresh_token: refreshToken,
+    }).pipe(
+      tap((response: AuthRefreshTokenResponse): void => {
+        localStorage.setItem('accessToken', response.id_token);
+      })
+    );
+  }
+
   logout(): void {
     this.setToken(null);
+    localStorage.clear();
```

Comment:
if you call setToken(null) there you call localStorage.clear(), you don't need cal it here